src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1507,6 +1512,38 @@ void NetworkOPsImp::consensusViewChange ()
         setMode (omCONNECTED);
 }
 
+void NetworkOPsImp::pubManifest (Manifest const& mo)
+{
+    // VFALCO consider std::shared_mutex
+    ScopedLockType sl (mSubLock);
+
+    if (!mSubManifests.empty ())
+    {
+        Json::Value jvObj (Json::objectValue);
+
+        jvObj [jss::type]        = "manifestReceived";
+        jvObj [jss::master_key]  = toBase58(TokenType::TOKEN_NODE_PUBLIC, mo.masterKey);
+        jvObj [jss::signing_key] = toBase58(TokenType::TOKEN_NODE_PUBLIC, mo.signingKey);
+        jvObj [jss::seq]         = Json::UInt (mo.sequence);
+        jvObj [jss::signature]   = strHex (mo.getSignature ());
+
+        for (auto i = mSubManifests.begin (); i != mSubManifests.end (); )
+        {
+            InfoSub::pointer p = i->second.lock ();
+
+            if (p)
```

Comment:
Nit: I prefer `if (auto p = i->second.lock())` to reduce the scope of `p` to where it's needed. This is a nit, you can leave the code as is if you want.